src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/OracleEngine.java

```diff
@@ -230,113 +231,146 @@ public synchronized void flush() throws DatabaseEngineException {
+                            } else {
+                                ps.setBytesForBlob(i, valArray);
+                            }
+                            break;
+                        case JSON:
+                        case CLOB:
+                            if (val == null) {
+                                ps.setNull(i, Types.CLOB);
+                                break;
+                            }
+                            if (val instanceof String) {
+                                final String valString = (String) val;
+                                if (fromBatch && valString.length() > MIN_SIZE_FOR_CLOB) {
+                                    // Use a clob for columns greater than 35K when inside a batch
+                                    // update because the Oracle driver creates one and only releases
+                                    // it when the PreparedStatement is closed. This will be closed
+                                    // explicitly when the batch is flushed.
+                                    final Clob clob = ps.getConnection().createClob();
+                                    clob.setString(1, valString);
+                                    ps.setClob(i, clob);
+                                    this.postFlushActions.add(clob::free);
+                                } else {
+                                    ps.setStringForClob(i, valString);
+                                }
+                            } else {
+                                throw new DatabaseEngineException("Cannot convert " + val.getClass().getSimpleName() + " to String. CLOB columns only accept Strings.");
+                            }
+                            break;
+                        case BOOLEAN:
+                            Boolean b = (Boolean) val;
+                            if (b == null) {
+                                ps.setObject(i, null);
+                            } else if (b) {
+                                ps.setObject(i, "1");
                             } else {
-                                ps.setStringForClob(i, valString);
+                                ps.setObject(i, "0");
                             }
-                        } else {
-                            throw new DatabaseEngineException("Cannot convert " + val.getClass().getSimpleName() + " to String. CLOB columns only accept Strings.");
+
+                            break;
+                        case DOUBLE:
+                            setParameterValues(ps, i, val);
+                            break;
+                        default:
+                            ps.setObject(i, ensureNoUnderflow(val));
+                    }
+                    i++;
+                } catch (final SQLException ex) {
+                    this.logger.debug("Something went wrong persisting the entity '{}'", entity.getName());
+                    if (ex.getErrorCode() == 8103) {
```

Comment:
We want to retry to setup the PreparedStatement if Oracle gives us this error:
SQLException: ORA-08103: object no longer exists